components/remerge/src/storage/info.rs

```diff
@@ -0,0 +1,307 @@
+        let mut fields = JsonObject::default();
+        // Note: we should probably report special telemetry for many of these
+        // errors, as they indicate (either a bug in remerge or)
+        for native_field in &self.native.fields {
+            // First try the record. Note that the `name` property isnt'
+            // supposed to change, barring removal or similar. (This is why
+            // `local_name` exists)
+            if let Some(value) = record.get(&native_field.name) {
+                fields.insert(native_field.local_name.clone(), value.clone());
+                continue;
+            } else if let Some(default) = native_field.ty.get_default() {
+                // Otherwise, we see if the field has a default value specified
+                // in the native schema.
+                fields.insert(native_field.local_name.clone(), default);
+                continue;
+            }
+            // If not, see if it has a default specified in the local schema.
+            // Even though we apply defaults when writing local records into the
+            // DB, this can happen if the local schema we wrote `record` with is
+            // an older version than our current local schema version.
+            if let Some(default) = self
+                .local
+                .field(&native_field.name)
+                .and_then(|lf| lf.ty.get_default())
+            {
+                // Make sure that that default is valid. If it's not, we
+                // ignore it (unless it's a required native field, in which
+                // case we complain).
+                if let Ok(fixed) = native_field.validate(default.clone()) {
+                    if fixed == default {
+                        fields.insert(native_field.local_name.clone(), default);
+                        continue;
+                    }
+                    // If this is actually a problem (e.g. the field is
+                    // required), we'll complain loudly below (this is likely a
+                    // schema issue if the field is required).
+                    log::error!(
+                        "More recent schema has default record for field {:?}, but it required fixups according to the native schema!",
+                        native_field.local_name,
+                    );
+                } else {
+                    // Same -- this is bad, but if it really matters we report
+                    // below.
+                    //
+                    // (That said, for this one, it might be worth always
+                    // erroring for tho, since the local schema's default value
+                    // for some field being invalid according to the native
+                    // schema should be a breaking change)
+                    log::error!(
+                        "More recent schema has default record for field {:?}, but it required fixups according to the native schema!",
```

Comment:
Nit: maybe this error message should differ from the previous one, something more like `was not valid according to the native schema`.